### PR TITLE
fix: remove existing zfs pool

### PR DESCRIPTION
**Pull request by jfroche**

### CODE REVIEW

**Comment by Lassulus** [comment]
maybe we can check if zpool is in PATH (like with mdadm) but maybe thats also a bad heuristic

**Comment by Lassulus** [comment]
I should probably extend PATH instead of overriding it at the top, hmm

File: types.nix [hunk 1]
```diff
@@ -164,6 +164,10 @@ rec {
         umount -Rlv /mnt
       fi
 
+      for ds in $(zpool list -H -o name); do
+        zpool destroy "$ds"
+      done
```
Comment (Mic92) [inline types.nix:169]:
```suggestion
      if command -v zpool; then
        for ds in $(zpool list -H -o name); do
          zpool destroy "$ds"
        done
      fi
```

File: types.nix [hunk 2]
```diff
@@ -164,6 +164,10 @@ rec {
...
+
       # stop all existing raids
       if command -v mdadm; then
```
Comment (Mic92) [inline types.nix:172]:
```suggestion
      if command -v mdadm &>/dev/null; then
```

File: types.nix (hunk 1 above)
Comment (zimbatm) [reply types.nix:169]: `s/command -v/type -p/g` because it's shorter :-p

**Comment by Lassulus** [comment]
I was inspired by this PR to do this: https://github.com/nix-community/disko/pull/75

**Comment by jfroche** [comment]
ok closing this one. I tested #75 and added https://github.com/nix-community/disko/pull/79 to make it work for me

**Pull request closed without merging**